### ISSUE TITLE
Improve page titles for Resource Docs pages

### ISSUE THREAD

**Issue by lukehoban** [pulumi/docs#4629]
The titles on resource docs pages feel like they could be better - they in practice always feel a little awkward and hard to parse, which could well have a non-trivial negative impact on whether these get strong SEO placement and engagement in Google/etc.

I wonder if instead of `Resource Instance | Module rds | Package AWS | Pulumi` we can make this:
* `aws.rds.Instance | Pulumi` or
* `AWS RDS Instance | Pulumi`

![Screen Shot 2020-11-20 at 10 54 17 AM](https://user-images.githubusercontent.com/223467/99841241-b3c75680-2b22-11eb-9f25-c2261de8efac.png)

**Comment by pgavlin** [comment]
@komalali can you take a look at this?

**Comment by lukehoban** [comment]
Cc also @davidwrede and @zchase
